cadl-tests/src/main/java/com/cadl/builtin/implementation/BuiltinClientImpl.java

```diff
@@ -133,6 +134,8 @@ public interface BuiltinClientService {
         @UnexpectedResponseExceptionType(HttpResponseException.class)
         Mono<Response<BinaryData>> read(
                 @HostParam("endpoint") String endpoint,
+                @QueryParam("query") String queryParam,
+                @QueryParam(value = "query-encoded", encoded = true) String queryParamEncoded,
```

Comment:
This would be vanilla logic. If query parameter is already encoded, use the `encoded = true`.